src/k2/persistence/logStream/LogStream.cpp

```diff
@@ -0,0 +1,547 @@
+                                for (auto& b: payload.shareAll().release()) {
+                                    read_payload.appendBinary( std::move(b));
+                                }
+                                return seastar::make_ready_future<>();
+                            });
+                        }
+                    )
+                    .then([&] (){
+                        return seastar::make_ready_future<Payload>(std::move(read_payload));
+                    });
+                });
+            })
+            .then([&] (auto&& payload){
+                std::unordered_map<Verb, std::vector<dto::MetadataRecord> > logStreamRecords;
+                Verb logStreamName;
+                uint32_t offset;
+                String plogId;
+                payload.seek(0);
+                while (payload.getDataRemaining() > 0){
+                    payload.read(logStreamName);
+                    payload.read(offset);
+                    payload.read(plogId);
+                    auto it = logStreamRecords.find(logStreamName);
+                    if (it == logStreamRecords.end()) {
+                        std::vector<dto::MetadataRecord> log;
+                        dto::MetadataRecord element{.plogId=plogId, .sealed_offset=0};
+                        log.push_back(std::move(element));
+                        logStreamRecords[logStreamName] = std::move(log);
+                    }
+                    else{
+                        it->second.back().sealed_offset = offset;
+                        dto::MetadataRecord element{.plogId=plogId, .sealed_offset=0};
+                        it->second.push_back(std::move(element));
+                    }
+                }
+
+                return seastar::do_with(std::move(logStreamRecords), [&] (auto& logStreamRecords){
+                    std::vector<seastar::future<std::tuple<Status, std::tuple<uint32_t, bool> > > > getStatusFutures;
+                    for (int logstreamName=LogStreamType::LogStreamTypeHead+1; logstreamName<LogStreamType::LogStreamTypeEnd; ++logstreamName){
+                        getStatusFutures.push_back(_logStreamMap[logstreamName]->get_plog_status(logStreamRecords[logstreamName].back().plogId));
+                    }
+                    return seastar::when_all_succeed(getStatusFutures.begin(), getStatusFutures.end())
+                    .then([&] (auto&& responses){
+                        Verb logstreamName=LogStreamType::LogStreamTypeHead;
+                        std::vector<seastar::future<Status> > reloadFutures;
+                        for (auto& response: responses){
+                            auto& [status, resp] = response;
+                            if (!status.is2xxOK()) {
+                                throw k2::dto::PlogGetStatusError("unable to obtain the information of a plog");
+                            }
+                            ++logstreamName;
```

Comment:
It seems very confusing to see a ++ on a variable called "*name".  Is this not a type? If so, I'd rather we have a helper method which recovers a particular log type, and we call that from the global method for each type we want.

Reply:
changed all the verbs to K2_DEF_ENUM